File: 프로그래머스 크레인 인형 뽑기.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> s;
    int i;
    for(i=0; i<moves.size(); i++)
    {
        int idx = 0;
        while(1)
        {
            if(idx == board.size())
                break;
            int temp = board[idx][moves[i]-1];
            if(temp != 0)
            {
                board[idx][moves[i]-1] = 0;
                if(s.size() == 0)
                {
                    s.push(temp);
                    break;
                }
                if(s.top() == temp)
                {
                    answer += 2;
                    s.pop();
                    break;
                }
                else
                {
                    s.push(temp);
                    break;
                }
            }
            idx++;
        }
    }
    return answer;
}

File: 프로그래머스 크레인 인형 뽑기.py
def solution(board, moves):
    answer = 0
    arr = []
    for i in moves:
        for j in range(0, len(board)):
            if board[j][i-1] != 0:
                temp = board[j][i-1]
                board[j][i-1] = 0
                if len(arr) == 0:
                    arr.append(temp)
                else:
                    if arr[len(arr)-1] == temp:
                        arr.pop(len(arr)-1)
                        answer += 2
                    else:
                        arr.append(temp)
                break
    return answer